drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -2,63 +2,95 @@ local capabilities = require "st.capabilities"
-  SinglePrecisionFloat(0, 8, 0.953125),
-  SinglePrecisionFloat(0, 9, 0.171875),
-  SinglePrecisionFloat(0, 9, 0.3671875),
-  SinglePrecisionFloat(0, 9, 0.5625),
-  SinglePrecisionFloat(0, 9, 0.7578125),
-  SinglePrecisionFloat(0, 9, 0.953125),
-
-  SinglePrecisionFloat(0, 10, 0.07421875),
-  SinglePrecisionFloat(0, 10, 0.171875),
-  SinglePrecisionFloat(0, 10, 0.26953125),
-  SinglePrecisionFloat(0, 10, 0.3671875),
-  SinglePrecisionFloat(0, 10, 0.46484375),
-  SinglePrecisionFloat(0, 10, 0.5625),
-  SinglePrecisionFloat(0, 10, 0.66015625),
-  SinglePrecisionFloat(0, 10, 0.7578125),
-  SinglePrecisionFloat(0, 10, 0.85546875),
-  SinglePrecisionFloat(0, 10, 0.953125),
-
-  SinglePrecisionFloat(0, 11, 0.025390625),
-  SinglePrecisionFloat(0, 11, 0.07421875),
-  SinglePrecisionFloat(0, 11, 0.123046875)
+local preference_map = {
+  [restorePowerState.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = RESTORE_POWER_STATE_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Boolean,
+  },
+  [changeToWirelessSwitch.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = CHANGE_TO_WIRELESS_SWITCH_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Uint8,
+    value_map = { [true] = 0x00,[false] = 0x01 },
+  },
+  [maxPower.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = MAX_POWER_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.SinglePrecisionFloat,
+    value_map = {
+      ["1"] = SinglePrecisionFloat(0, 6, 0.5625),
+      ["2"] = SinglePrecisionFloat(0, 7, 0.5625),
+      ["3"] = SinglePrecisionFloat(0, 8, 0.171875),
+      ["4"] = SinglePrecisionFloat(0, 8, 0.5625),
+      ["5"] = SinglePrecisionFloat(0, 8, 0.953125),
+      ["6"] = SinglePrecisionFloat(0, 9, 0.171875),
+      ["7"] = SinglePrecisionFloat(0, 9, 0.3671875),
+      ["8"] = SinglePrecisionFloat(0, 9, 0.5625),
+      ["9"] = SinglePrecisionFloat(0, 9, 0.7578125),
+      ["10"] = SinglePrecisionFloat(0, 9, 0.953125),
+      ["11"] = SinglePrecisionFloat(0, 10, 0.07421875),
+      ["12"] = SinglePrecisionFloat(0, 10, 0.171875),
+      ["13"] = SinglePrecisionFloat(0, 10, 0.26953125),
+      ["14"] = SinglePrecisionFloat(0, 10, 0.3671875),
+      ["15"] = SinglePrecisionFloat(0, 10, 0.46484375),
+      ["16"] = SinglePrecisionFloat(0, 10, 0.5625),
+      ["17"] = SinglePrecisionFloat(0, 10, 0.66015625),
+      ["18"] = SinglePrecisionFloat(0, 10, 0.7578125),
+      ["19"] = SinglePrecisionFloat(0, 10, 0.85546875),
+      ["20"] = SinglePrecisionFloat(0, 10, 0.953125),
+      ["21"] = SinglePrecisionFloat(0, 11, 0.025390625),
+      ["22"] = SinglePrecisionFloat(0, 11, 0.07421875),
+      ["23"] = SinglePrecisionFloat(0, 11, 0.123046875)
+    },
+  },
+  [electricSwitchType.ID] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = ELECTRIC_SWITCH_TYPE_ATTRIBUTE_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Uint8,
+    value_map = { ['rocker'] = 0x01,['rebound'] = 0x02 },
```

Comment:
If your keys are strings, they don't need to be wrapped in brackets

```suggestion
    value_map = { rocker = 0x01, rebound = 0x02 },
```

Reply:
Will do.